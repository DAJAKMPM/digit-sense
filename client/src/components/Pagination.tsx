import React from "react";
import { Box, IconButton, useTheme } from "@mui/material";
import KeyboardArrowLeft from "@mui/icons-material/KeyboardArrowLeft";
import KeyboardArrowRight from "@mui/icons-material/KeyboardArrowRight";
import { PaginationProps } from "../types";

const Pagination = ({
  count,
  page,
  rowsPerPage,
  onPageChange,
}: PaginationProps) => {
  const theme = useTheme();

  const handleBackButtonClick = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    onPageChange(event, page - 1);
  };

  const handleNextButtonClick = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    onPageChange(event, page + 1);
  };

  return (
    <Box sx={{ flexShrink: 0, ml: 2.5 }}>
      <IconButton
        onClick={handleBackButtonClick}
        disabled={page === 0}
        aria-label="previous page"
      >
        {theme.direction === "rtl" ? (
          <KeyboardArrowRight />
        ) : (
          <KeyboardArrowLeft />
        )}
      </IconButton>
      <IconButton
        onClick={handleNextButtonClick}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="next page"
      >
        {theme.direction === "rtl" ? (
          <KeyboardArrowLeft />
        ) : (
          <KeyboardArrowRight />
        )}
      </IconButton>
    </Box>
  );
};

export { Pagination };
