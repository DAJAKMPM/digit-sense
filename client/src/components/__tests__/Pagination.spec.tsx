import { render } from "@testing-library/react";
import { Pagination } from "../Pagination";

test("should match the Pagination snapshot properly", () => {
  const { container } = render(
    <Pagination count={5} page={1} rowsPerPage={5} onPageChange={() => {}} />
  );

  expect(container).toMatchSnapshot();
});
