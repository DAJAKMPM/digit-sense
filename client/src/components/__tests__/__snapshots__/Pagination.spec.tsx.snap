// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the Pagination snapshot properly 1`] = `
<div>
  <div
    class="MuiBox-root css-1zye22"
  >
    <button
      aria-label="previous page"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="next page"
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
      </svg>
    </button>
  </div>
</div>
`;
